classes/services/PKPSubmissionService.inc.php

```diff
@@ -795,6 +795,12 @@ public function delete($submission) {
 	public function canEditPublication($submissionId, $userId) {
 		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
 		$stageAssignments = $stageAssignmentDao->getBySubmissionAndUserIdAndStageId($submissionId, $userId, null)->toArray();
+		$submission = $this->get($submissionId);
```

Comment:
Every case where `canEditPublication` is called, the `$submission` object is already available -- so I'd suggest changing calling code to just pass the submission object instead of `$submission->getId()`. That would save the additional (expensive) fetch of the submission object from the DB.

Reply:
Hi, @asmecher ! :smiley:   
I've [adjusted](https://github.com/pkp/pkp-lib/pull/10693/commits/4a084361365b6d8fd0875846f03d3ca102688c7e) the `canEditPublication` method and updated the calls in lib-pkp. I've also done the PR to include the method call update in [OMP](https://github.com/pkp/omp/pull/1786) and [OPS](https://github.com/pkp/ops/pull/826).

Reply:
@YvesLepidus I think you're missing a PR for OJS too, no?

Reply:
I couldn't find a call to the `canEditPublication` method in the [OJS 3.3.0 code](https://github.com/search?q=repo%3Apkp%2Fojs%20canEditPublication&type=code). :thinking: 
I only [found](https://github.com/pkp/ojs/blob/edd0f343981158db790dd6779ea4fc4f2cda7238/pages/workflow/WorkflowHandler.php#L122) it in the `main` branch, using Repository.